warehouse/utils/utils.go

```diff
@@ -230,6 +230,7 @@ type KeyValue struct {
 	Value interface{}
 }
 
+//go:generate mockgen -destination=../internal/mocks/utils/mock_uploader.go -package mock_uploader github.com/rudderlabs/rudder-server/warehouse/utils Uploader
```

Comment:
Can we move this in a separate package?

Reply:
I had created already a dedicated package for it, namely `warehouse/internal/mocks/utils/mock_uploader.go`. I was planning to have other mocks there as well. What's wrong with it?

Reply:
So what's your suggestion? What's the exact package name that you'd create for hosting this mock?

Reply:
The mock package name is fine, I was simply asking if we can move this interface to a separate package `warehouse/uploader`?

Reply:
Right, let's move it as part of another cleanup PR where we refactor this a bit. There are several things we want to address, splitting the interface in a more SOLID approach, move the derivated interfaces in appropriate packages and remove the circular references. We can do it all together as a separate change, wdyt?

Reply:
Sure.